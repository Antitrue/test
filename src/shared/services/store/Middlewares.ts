import { authAPI } from '../';
import { userCommentAPI } from '../api/user/userComment.api';
import { clientReviewAPI } from '../api/client/clientReview.api';
import { petContactAPI } from '../api/client/petContact.api';
import { clientNewsAPI } from '../api/client/clientNews.api';
import { topicApi } from '../api/user/topic.api';
import { reproductionAPI } from '../api/client/reproduction.api.ts';

export const middlewares = [
  authAPI.middleware,
  clientNewsAPI.middleware,
  petContactAPI.middleware,
  clientReviewAPI.middleware,
  userCommentAPI.middleware,
  topicApi.middleware,
  reproductionAPI.middleware,
];
