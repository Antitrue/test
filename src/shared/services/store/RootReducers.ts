import { combineReducers } from '@reduxjs/toolkit';
import { authAPI, authenticationSlice } from '../';
import { userCommentAPI } from '../api/user/userComment.api';
import { clientReviewAPI } from '../api/client/clientReview.api';
import { petContactAPI } from '../api/client/petContact.api';
import { clientNewsAPI } from '../api/client/clientNews.api';
import { topicApi } from '../api/user/topic.api';

const rootReducers = combineReducers({
  authenticationSlice: authenticationSlice.reducer,
  [authAPI.reducerPath]: authAPI.reducer,
  [userCommentAPI.reducerPath]: userCommentAPI.reducer,
  [clientReviewAPI.reducerPath]: clientReviewAPI.reducer,
  [petContactAPI.reducerPath]: petContactAPI.reducer,
  [clientNewsAPI.reducerPath]: clientNewsAPI.reducer,
  [topicApi.reducerPath]: topicApi.reducer,
});

export default rootReducers;
