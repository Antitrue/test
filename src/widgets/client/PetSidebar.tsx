import type { MenuProps } from 'antd';
import { Flex, Menu, Button, ConfigProvider } from 'antd';
import cn from 'classnames';
import { v4 as uuidv4 } from 'uuid';
import { useGetAllPetsQuery } from '../../shared/services/api/client/petClient.api';

import styles from './PetSidebar.module.scss';
import { useMemo } from 'react';

type MenuItem = Required<MenuProps>['items'][number];

const PetSidebar: React.FC = () => {
  const { data = [], isError, isLoading } = useGetAllPetsQuery();

  const items: MenuItem[] = useMemo(() => {
    return data.map((pet, index: number) => {
      const buttons = (
        <div>
          <a href={`/${pet.name}`} className={styles.sidebar__edit_link}>
            <Button type='primary' className={cn(styles.sidebar__buttons, styles.sidebar__edit_button)}>
              Редактировать
            </Button>
          </a>
          <Button
            type='primary'
            onClick={() => {
              console.log(`${pet.name} deleted`);
            }}
            danger
            className={cn(styles.sidebar__buttons, styles.sidebar__delete_button)}>
            Удалить
          </Button>
        </div>
      );
      const iconUrl =
        typeof pet.avatar === 'string'
          ? pet.avatar
          : 'https://static-00.iconduck.com/assets.00/animal-icon-2048x2048-rdj39f2s.png';
      return {
        key: uuidv4(),
        label: pet.name,
        icon: <img src={iconUrl} alt={pet.name} />,
        children: [{ key: `${index}${index}`, label: buttons }],
      };
    });
  }, [data]);

  return (
    <Flex vertical className={styles.sidebar}>
      <Flex className={styles.sidebar__header}>
        <span className={styles.sidebar__header_text}>Ваши питомцы</span>
        <button type='button' className={styles.sidebar__header_button}></button>
      </Flex>
      <ConfigProvider
        theme={{
          components: {
            Menu: {
              itemHoverBg: 'rgba(84, 150, 212, 0.2)',
              itemColor: 'rgba(58, 64, 69, 1)',
            },
          },
        }}>
        {isLoading ? <div>Loading...</div> : null}
        {isError ? (
          <div>Pets data are having error</div>
        ) : (
          <Menu selectable={false} className={styles.sidebar__menu} mode='inline' items={items} />
        )}
      </ConfigProvider>
    </Flex>
  );
};

export default PetSidebar;
