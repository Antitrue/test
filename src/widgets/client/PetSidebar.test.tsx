import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import PetSidebar from './PetSidebar';
import { Provider } from 'react-redux';
import { store } from '../../shared/services/store/store';

describe('PetSidebar component', () => {
  it('PetSidebar renders', () => {
    const { baseElement } = render(
      <Provider store={store}>
        <PetSidebar />
      </Provider>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
