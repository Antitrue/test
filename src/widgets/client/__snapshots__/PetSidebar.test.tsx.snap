// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetSidebar component PetSidebar renders 1`] = `
<body>
  <div>
    <div
      class="ant-flex css-dev-only-do-not-override-1c0na6j ant-flex-align-stretch ant-flex-vertical"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-1c0na6j"
      >
        <span>
          Ваши питомцы
        </span>
        <button
          type="button"
        />
      </div>
      <div>
        Loading...
      </div>
      <ul
        class="ant-menu ant-menu-root ant-menu-inline ant-menu-light css-dev-only-do-not-override-18dsc0v"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      />
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </div>
  </div>
</body>
`;
